core/src/main/java/io/grpc/internal/AbstractClientStream.java

```diff
@@ -119,8 +124,19 @@ protected void inboundHeadersReceived(Metadata.Headers headers) {
     }
     if (headers.containsKey(ChannelImpl.MESSAGE_ENCODING_KEY)) {
       String messageEncoding = headers.get(ChannelImpl.MESSAGE_ENCODING_KEY);
-      setDecompressor(messageEncoding);
+      try {
+        setDecompressor(messageEncoding);
+      } catch (IllegalArgumentException e) {
+        // Don't use INVALID_ARGUMENT since that is for servers to send clients.
+        Status status = Status.INTERNAL.withDescription(e.getMessage()).withCause(e);
```

Comment:
We haven't generally been copying the message from caused exceptions. However, we should really be providing more context here anyway, so can we replace `e.getMessage()` with a more clear message. Maybe something like: "Unable to decompress messages from server".